README.md

```diff
@@ -0,0 +1,2469 @@
+
+We define `Events` for a given class
+that we want to manage. 
+In the example above,
+we are managing *Pets*.
+We have three possible events,
+in this scenario.
+
+We can call `petBloc.add(event)` 
+(which is a **`bloc`**)
+and pass an instance of the event to the `bloc`.
+
+The `bloc` can do some logic inside.
+For example, it makes an API call
+or accesses a service.
+
+Afterwards, the `bloc` **emits a `state`**.
+
+### [Cubit](https://bloclibrary.dev/#/coreconcepts?id=cubit) 
+
+A `Cubit` is a **much simpler, minimalistic version of a `Bloc`**.
+Unlike `Bloc`,
+the `Cubit` exposes functions that can be invoked
+to trigger state changes.
+
+Check the following diagram.
+
+<img width="1127" alt="cubit-diagram" src="https://user-images.githubusercontent.com/17494745/223120206-dee22295-94f4-472a-803d-45ca9f3cab45.png">
+
+> credits of the image go to https://www.youtube.com/watch?v=sAz_8pRIf5E&ab_channel=BradCypert.
+
+`Cubits`, 
+although similar,
+differ from `Blocs` because
+**they don't have events**.
+The `Cubit` has *methods*,
+there is no need to pass instances of `events`
+like we do in `Blocs`.
+
+Inside these methods you would write your logic
+(making a call to an API, for example),
+and inside the same method you would also emit state.
+
+
+### Wait... so which to use?
+
+You may be wondering yourself
+"Which one should I choose?".
+As always, the answer is... **it depends**.
+
+|    |       **Cubit**      |              **Bloc**              |
+|:---------------:|:-----------:|:----------------------------------:|
+| **Simplicity**                     | <ul><li>states</li><li>functions</li></ul> | <ul><li>states</li><li>events</li><li>event handlers</li></ul> |
+| **Traceability**                   |    | Transition (onTransition())      |
+| **Advanced Event Transformations** |    | EventTransformer                   |
+
+`Cubit` shines with its simplicity
+and it's better suited for simple use cases.
+If your team is struggling to model event transitions,
+you might want to start with `Cubits`.
+
+However, if you value traceability 
+or are having a hard time mocking cubits for widget tests,
+you might want to use `Blocs`.
+
+If you're unsure of the event-driven approach,
+start with `Cubit`.
+You can always refactor the code later on 
+to a `Bloc` when you have a clearer idea
+of the possible events of your application.
+
+If you want to see a more in-depth
+comparison between the two, 
+please visit
+https://bloclibrary.dev/#/coreconcepts?id=cubit-vs-bloc.
+
+
+## `BLoC` concepts in Flutter 🦋
+
+Now that we are aware of the most important `Bloc` concepts,
+let's see how these could be implemented in Flutter.
+
+### Defining a `Bloc`
+
+To create a `bloc` in Flutter,
+we need to:
+- provide an initial state.
+- set up event listeners and handlers.
+- add events to `bloc` via `bloc.add`.
+
+Check the following piece of code.
+
+```dart
+class PetBloc extends Bloc<PetEvent, PetState> {
+    final PetRepo repo;
```

Comment:
what is the `repo`? 
(please add a comment linking docs to where people can understand this boilerplate code ... 🔗 🙏 )
e.g: https://bloclibrary.dev/#/architecture?id=repository
But documentation uses "`repository`" ... looks like it's just a variable name. 
maybe we should use the same one for consistency. 💭